pkg/cache/tas_flavor_snapshot.go

```diff
@@ -226,6 +226,22 @@ func (s *TASFlavorSnapshot) addTASUsage(domainID utiltas.TopologyDomainID, usage
 	s.leaves[domainID].tasUsage.Add(usage)
 }
 
+func (s *TASFlavorSnapshot) FreeCapacityPerDomain() map[string]int {
+	freeCapacityPerDomain := make(map[string]int)
+
+	for domainID, leaf := range s.leaves {
+		// Calculate the total free capacity for the domain
+		totalFreeCapacity := 0
+		for _, quantity := range leaf.freeCapacity {
+			totalFreeCapacity += int(quantity)
```

Comment:
What does this number mean? IIUC we are flattening heterogenous resources (cpu/gpu/memory) into a single quantity. I don't think we want to lose this granularity.